File: guides/micronaut-email/java/src/main/java/example/micronaut/EmailCmd.java
package example.micronaut;

import io.micronaut.core.annotation.Introspected;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.List;

//tag::clazzwithannotations[]
@EmailConstraints
//end::clazzwithannotations[]
//tag::clazz[]
@Introspected
public class EmailCmd implements Email {
//end::clazz[]

    //tag::properties[]
    @NotNull
    @NotBlank
    private String recipient;

    @NotNull
    @NotBlank
    private String subject;

    private List<String> cc = new ArrayList<>();
    private List<String> bcc = new ArrayList<>();

    private String htmlBody;
    private String textBody;
    private String replyTo;
    //end::properties[]

    //tag::settersandgetters[]
    @Override
    public String getRecipient() {
        return recipient;
    }

    public void setRecipient(String recipient) {
        this.recipient = recipient;
    }

    @Override
    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    @Override
    public List<String> getCc() {
        return cc;
    }

    public void setCc(List<String> cc) {
        this.cc = cc;
    }

    @Override
    public List<String> getBcc() {
        return bcc;
    }

    public void setBcc(List<String> bcc) {
        this.bcc = bcc;
    }

    @Override
    public String getHtmlBody() {
        return htmlBody;
    }

    public void setHtmlBody(String htmlBody) {
        this.htmlBody = htmlBody;
    }

    @Override
    public String getTextBody() {
        return textBody;
    }

    public void setTextBody(String textBody) {
        this.textBody = textBody;
    }

    @Override
    public String getReplyTo() {
        return replyTo;
    }

    public void setReplyTo(String replyTo) {
        this.replyTo = replyTo;
    }
}
//end::settersandgetters[]
/*
//tag::close[]
    // getters & setters
}
//end::close[]
 */

File: guides/micronaut-email/kotlin/src/main/kotlin/example/micronaut/EmailCmd.kt
package example.micronaut

import io.micronaut.core.annotation.Introspected
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

//tag::clazzwithannotations[]
@EmailConstraints
//end::clazzwithannotations[]
//tag::clazz[]
@Introspected
class EmailCmd : Email {
//end::clazz[]

    //tag::properties[]
    @NotBlank
    @NotNull
    var recipient: String? = null

    @NotBlank
    @NotNull
    var subject: String? = null

    var cc: List<String>? = null

    var bcc: List<String>? = null

    var htmlBody: String? = null

    var textBody: String? = null

    var replyTo: String? = null
    //end::properties[]

    //tag::settersandgetters[]
    override fun recipient(): String? {
        return this.recipient
    }

    override fun cc(): List<String>? {
        return this.cc
    }

    override fun bcc(): List<String>? {
        return this.bcc
    }

    override fun subject(): String? {
        return this.subject
    }

    override fun htmlBody(): String? {
        return this.htmlBody
    }

    override fun textBody(): String? {
        return this.textBody
    }

    override fun replyTo(): String? {
        return this.replyTo
    }
}
//end::settersandgetters[]
/*
//tag::close[]
    // getters & setters
}
//end::close[]
 */

